src/components/OptionsBar/Options/FilterModal/Filters/RelativeFilters.jsx

```diff
@@ -0,0 +1,48 @@
+import { useSelector } from 'react-redux';
+import { useRef } from 'react';
+
+const relativeFilters = (filterIds, tabIndex) => {
+  const filters = useSelector((state) => state.filters);
+
+  const getFilterMatrix = () => {
+    const filterMatrix = new Array(filterIds.length);
+
+    filterIds.forEach((id, index) => {
+      filterMatrix[index] = filters[id];
+    });
```

Comment:
Instead of creating an empty array, iterate, each element inject a value from another array; you can map over the filterIds, and the result will be just the same

```js
const filterMatrix = filters ? filterIds.map(id => filters?.[id]) : []
```

make sense? It also validates `filters`